dask/array/tests/test_array_core.py

```diff
@@ -3231,3 +3231,89 @@ def test_meta(dtype):
     assert a._meta.dtype == a.dtype
     assert isinstance(a._meta, np.ndarray)
     assert a.nbytes < 1000
+
+
+def test_zarr_roundtrip():
+    pytest.importorskip('zarr')
+    with tmpdir() as d:
+        a = da.zeros((3, 3), chunks=(1, 1))
+        a.to_zarr(d)
+        a2 = da.from_zarr(d)
+        assert_eq(a, a2)
+        assert a2.chunks == a.chunks
+
+
+def test_zarr_rechunk():
+    pytest.importorskip('zarr')
+    a = da.zeros((3, ), chunks=((1, 2), ))
+    with tmpdir() as d:
+        with pytest.raises(ValueError):
+            a.to_zarr(d)
+        a.to_zarr(d, rechunk=(1, ))
+
+
+def test_read_zarr_chunks():
+    pytest.importorskip('zarr')
+    a = da.zeros((9, ), chunks=(3, ))
+    with tmpdir() as d:
+        a.to_zarr(d)
+        arr = da.from_zarr(d, chunks=(5, ))
+        assert arr.chunks == ((5, 4), )
```

Comment:
Question: Should we raise or warn here due to the mismatched chunking?

Reply:
The last chunk can always be smaller, this is fine. The only time this could conceivably be an issue is if concatenating along that axis.

Reply:
I'm actually concerned about the 3->5 change.  A user should probably be encouraged to choose a chunking that is a mulitple of 3.

Reply:
I'd say that the option is provided for the user to choose a chunking that makes sense for what they are doing. We wouldn't warn if the user had called `.rechunk()` directly with whatever specific chunkset.
In any case, gauging whether a rechunk operation is expensive is tricky. In this case, the operation is clearly not required at all, but when would you warn if the input was `((3, 4, 4, 4),)`?  the best we could really do is look at the relative sizes of the graphs.  I would be OK to warn if the size of the final graph is, say, >2x the input one.